prov/shm/src/smr_msg.c

```diff
@@ -420,23 +422,23 @@ static ssize_t smr_generic_inject(struct fid_ep *ep_fid, const void *buf,
 		return -FI_EAGAIN;
 
 	ret = smr_cmd_queue_next(smr_cmd_queue(peer_smr), &ce, &pos);
-	if (ret == -FI_ENOENT)
-		return -FI_EAGAIN;
+	if (ret == -FI_ENOENT) {
+		ret = -FI_EAGAIN;
+		goto signal;
+	}
 
 	proto = len <= SMR_MSG_DATA_LEN ? smr_src_inline : smr_src_inject;
 	ret = smr_proto_ops[proto](ep, peer_smr, id, peer_id, op, tag, data,
 			op_flags, NULL, &msg_iov, 1, len, NULL, &ce->cmd);
 	if (ret) {
 		smr_cmd_queue_discard(ce, pos);
-		return -FI_EAGAIN;
```

Comment:
do you need to `ret = -FI_EAGAIN` here?

Reply:
I don't think so.  If the protocol function returns FI_EAGAIN, then we pass that up the stack, but maybe the protocol function has a different error.  In that case a signal isn't necessary (but doesn't hurt) and we don't hide it behind EAGAIN.